### ISSUE TITLE
AvalancheGo went into a state that caused intermittent failing health checks until it was restarted.

### ISSUE THREAD

**Issue by CarnivalBen** [ava-labs/avalanchego#1286]
**Describe the bug**
Logging this as an issue at the request of laviniadenisat on discord: https://discord.com/channels/578992315641626624/620633143002660874/1092411534438826144

At exactly 2 hours before my three month stake came to an end, my node seemed to go into a strange state where it started failing health checks, and recovering, and repeating.

Looking at Grafana, I could see that database latency had gone from usually around 2ms, to between 20 and 40ms.

Looking at the logs, I can see some warnings similar to:

```
Apr 02 15:57:49 1079A5 avalanchego[1592853]: [04-02|15:57:47.618] WARN <C Chain> handler/handler.go:453 handling sync message took longer than expected {"processingTime": "31.041879945s", "nodeID": "NodeID-L9HsJPQ9TF5aXjdx3KnQrh1eWaHJaNXNj", "messageOp": "put", "message": "chain_id:\"\\x04'Բ**x\\xbc\\xdd\\xd4Vt,\\xaf\\x91\\xb5k\\xad\\xbf\\xf9\\x85\\xee\\x19\\xae\\xf1Es\\xe74?\\xd6R\" request_id:4294967295 container:\"\\x00\\x00\\x00\\x00\\x00\\x00\\x16\\x98\\xbb
```

I restarted avalanchego (`sudo systemctl restart avalanchego`) around 4 hours after the problem started and since then it has gone back to normal, with no problems since.

**To Reproduce**
Unable to reproduce.

**Expected behavior**
Expect the node to continue to run with no errors/warnings.

**Logs**
I have attached the full logs covering the period.

[avalanchego-issue.txt](https://github.com/ava-labs/avalanchego/files/11137964/avalanchego-issue.txt)

Timestamps (UTC) to note:
```
Apr 02 15:57:49 - problem started
Apr 02 18:00:13 - staking period came to an end
Apr 02 18:09:02 - restaked for another 3 months (I hadn't noticed a problem at the time)
Apr 02 20:02:28 - avalanchego was restarted
```

**Screenshots**
I've attached some screenshots from grafana, showing the last 24 hours, with the problem occurring around 20 hours prior to the screenshots being taken, timezone: UTC+1:

<img width="925" alt="db-latency" src="https://user-images.githubusercontent.com/3448911/229507004-18044f0d-4261-4f1e-a72d-4d218b30fac0.png">

<img width="1867" alt="machine-metrics" src="https://user-images.githubusercontent.com/3448911/229507217-f0210dd7-4517-4f2d-b4f4-f2a66f39cbb8.png">

<img width="1876" alt="db-part1" src="https://user-images.githubusercontent.com/3448911/229507238-c4b8e389-91eb-4b3d-bbe7-06f6bd00f0de.png">

<img width="1865" alt="db-part2" src="https://user-images.githubusercontent.com/3448911/229507254-8e0b94f8-1299-4460-bd70-b3709bd8a859.png">

<img width="1851" alt="cchain-part1" src="https://user-images.githubusercontent.com/3448911/229507279-58b54f61-5200-41bf-bc2c-638ac6855d63.png">

<img width="1864" alt="cchain-part2" src="https://user-images.githubusercontent.com/3448911/229507301-83588ae4-2d04-4881-bfa0-239bf8880cb9.png">

**Metrics**
See screenshots from grafana.

**Operating System**
Ubuntu 20.04.4 LTS

**Additional context**
N/A

**To best protect the Avalanche community security bugs should be reported in accordance to our [Security Policy](/SECURITY.md)**

**Comment by aaronbuchwald** [comment]
Could you provide the full specs of the machine that you're running on as well?

**Comment by CarnivalBen** [comment]
Full specs:
AMD Ryzen 7 PRO 3700 8 core
64G RAM
2Tb SSD

**Comment by CarnivalBen** [comment]
Yes, it's exactly as you describe. I have not specifically configured anything around port 9650, so it's using the default settings. I can internally make api calls from the server itself (using curl), but all public api access is blocked via my hosting provider's firewall.

**Comment by StephenButtolph** [comment]
@CarnivalBen Is this computer being used for anything other than running avalanchego?

**Comment by CarnivalBen** [comment]
@StephenButtolph No, its a dedicated Avalanche Go server.

I'll just add that it hasn't happened again since restarting the node (using `sudo systemctl restart avalanchego`)

**Comment by CarnivalBen** [comment]
I'm starting to see this error again on my node.... its more intermittent than last time though. I can provide more logs if needed.

**Comment by joshua-kim** [comment]
Closing due to age, feel free to re-open if you see this again. Currently there is not enough information/repo available to investigate.
